skypy/halos/sham.py

```diff
@@ -0,0 +1,1234 @@
+
+    # Cosmology
+    line8 = 'cosmology: !astropy.cosmology.' + cosmology.name + '\n'
+    line9 = '  H0: ' + str(cosmology.h*100) + '\n'
+    line10 = '  Om0: ' + str(cosmology.Om0) + '\n'
+
+    # Call function
+    function = 'tables:\n  galaxy:\n'
+    function += '    z, sm: !skypy.galaxies.schechter_smf\n      redshift: $z_range\n'
+    function += '      m_star: $m_star\n      phi_star: $phi_star\n'
+    function += '      alpha: $alpha_val\n      m_min: $m_min\n'
+    function += '      m_max: $m_max\n      sky_area: $sky_area\n'
+
+    # Make one large string
+    yaml_lines = line1 + line2 + line3 + line4 + line5 + line6 + line7 + line8 + line9 + line10
+    yaml_lines += function
+
+    file_gal = open(file_name, 'w')
+    file_gal.write(yaml_lines)
+    file_gal.close()
+
+    # Execute file
+    return run_file(file_name, 'galaxy', 'sm')
+
+
+# Assignment
+def assignment(hs_order, rc_order, rs_order, bc_order, bs_order, qu_order, id_order, z_order,
+               scatter=False):
+    r'''Function that assigns galaxies to halos based on type and the
+    quenching function
+
+    This function runs through the halo catalogue, assesses whether it has a
+    quenched galaxy and if it is a central or satellite, and assigns the next
+    galaxy in the appropriate list. This therefore creates an ordered array
+    of halos and galaxies which are assigned to each other. Unassigned halos
+    and galaxies are deleted or ignored.
+
+    Parameters
+    -----------
+    hs_order : (nm, ) array_like
+        Ordered (most massive first) array of central and subhalo masses,
+        in units of solar mass
+    rc_order : (ng1, ) array_like
+        Ordered (most massive first) array of red central galaxy masses,
+        in units of solar mass
+    rs_order : (ng2, ) array_like
+        Ordered (most massive first) array of red satellite galaxy masses,
+        in units of solar mass
+    bc_order : (ng3, ) array_like
+        Ordered (most massive first) array of blue central galaxy masses,
+        in units of solar mass
+    bs_order : (ng4, ) array_like
+        Ordered (most massive first) array of blue satellite galaxy masses,
+        in units of solar mass
+    qu_order : (nm, ) array_like
+        Truth array of which halos should be assigned quenched (red) galaxies
+    id_order : (nm, ) array_like
+        Array of IDs to determine if the halo is central or satellite
+    z_order : (nm, ) array_like
+        Redshifts of generated halos
+    scatter : Boolean
+        True/false if the galaxies lists have undergone scatter
+
+    Returns
+    --------
+    hs_fin: (nh, ) array_like
+        List of assigned halo masses, in units of solar masses
+    gal_fin: (nh, ) array_like
+        List of assigned galaxies, in units of solar mass
+    id_fin: (nh, ) array_like
+        List of ID values for halos
+    z_fin: (nh, ) array_like
+        List of redshifts for halos
+    gal_type_fin: (nh, ) array_like

```

Comment:
Could this labels be more informative? i.e be strings like `'red_central'` rather than integers.